src/main/java/com/aws/greengrass/ipc/SubscribeToCertificateUpdatesOperationHandler.java

```diff
@@ -0,0 +1,190 @@
+    private static final String COMPONENT_NAME = "componentName";
+    private static final String NO_CERT_OPTIONS_ERROR = "Certificate options are required.";
+    private static final String NO_CERT_TYPE_ERROR = "Certificate type is required";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final int RSA_KEY_LENGTH_FOR_SERVER_CERT = 4096;
+    private final String serviceName;
+    private final ClientDevicesAuthService clientDevicesAuthService;
+    private final CertificateManager certificateManager;
+    private final AtomicBoolean subscriptionResponseSent = new AtomicBoolean(false);
+    private final AtomicReference<CertificateUpdateEvent> firstStreamingEvent = new AtomicReference<>(null);
+    private KeyPair keyPair;
+
+    /**
+     * Constructor.
+     *
+     * @param context                  operation continuation handler
+     * @param clientDevicesAuthService client device auth service
+     */
+    public SubscribeToCertificateUpdatesOperationHandler(OperationContinuationHandlerContext context,
+                                                         ClientDevicesAuthService clientDevicesAuthService) {
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.clientDevicesAuthService = clientDevicesAuthService;
+        this.certificateManager = clientDevicesAuthService.getCertificateManager();
+
+    }
+
+    @Override
+    protected void onStreamClosed() {
+        //TODO: Unsubscribe to cert updates.
+    }
+
+    /**
+     * Override to appropriately enforce stream responses are sent after the initial response.
+     * This only gets called if handleRequest returns normally and starts to send a response.
+     * <p>
+     * Here, we know that the initial SubscribeToCertificateUpdatesResponse response is sent. So, we send the
+     * first streaming response of CertificateUpdateEvent.
+     * </p>
+     */
+    @Override
+    public void afterHandleRequest() {
+        synchronized (firstStreamingEvent) {
+            if (subscriptionResponseSent.compareAndSet(false, true)) {
+                CertificateUpdateEvent event = firstStreamingEvent.get();
+                if (event != null) {
+                    sendStreamEvent(event);
+                }
+                firstStreamingEvent.set(null);
+            }
+        }
+    }
+
+    @SuppressWarnings("PMD.PreserveStackTrace")
+    @Override
+    public SubscribeToCertificateUpdatesResponse handleRequest(
+            SubscribeToCertificateUpdatesRequest subscribeToCertificateUpdatesRequest) {
+        return translateExceptions(() -> {
+            try {
+                clientDevicesAuthService.doAuthorizationForSubscribingToCertUpdates(serviceName);
+            } catch (AuthorizationException e) {
+                logger.atInfo().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
```

Comment:
There are a couple ways to look at what log level this should be. If you treat components as untrusted, and assume this would be a common occurrence, then info may make sense here. After all, it's not an error from CDAs perspective.

However, this really shouldn't happen under normal circumstances. I'd make this WARN